cmd/evm/runner.go

```diff
@@ -183,15 +197,30 @@ func runCmd(ctx *cli.Context) error {
 	}
 	tstart := time.Now()
 	var leftOverGas uint64
+
+	// sender account pre-pays gas
+	mgval := new(big.Int).Mul(new(big.Int).SetUint64(initialGas), runtimeConfig.GasPrice)
+	statedb.SubBalance(sender, mgval)
+	
 	if ctx.GlobalBool(CreateFlag.Name) {
 		input := append(code, common.Hex2Bytes(ctx.GlobalString(InputFlag.Name))...)
+		intrinsicGas := core.IntrinsicGas(input, ctx.GlobalBool(CreateFlag.Name), true)
+		initialGas -= intrinsicGas.Uint64()
+		runtimeConfig.TxGasLimit = initialGas
 		ret, _, leftOverGas, err = runtime.Create(input, &runtimeConfig)
 	} else {
 		if len(code) > 0 {
 			statedb.SetCode(receiver, code)
 		}
+		intrinsicGas := core.IntrinsicGas(common.Hex2Bytes(ctx.GlobalString(InputFlag.Name)), false, true)
+		initialGas -= intrinsicGas.Uint64()
+		runtimeConfig.TxGasLimit = initialGas
 		ret, leftOverGas, err = runtime.Call(receiver, common.Hex2Bytes(ctx.GlobalString(InputFlag.Name)), &runtimeConfig)
 	}
+	// refund sender unused prepaid gas
+	remainingEther := new(big.Int).Mul(new(big.Int).SetUint64(leftOverGas), runtimeConfig.GasPrice)
+	statedb.AddBalance(sender, remainingEther)
```

Comment:
Isn't this only necessary when `if ctx.GlobalBool(DumpFlag.Name)` is set, which does `stateDb.IntermediateRoot`?